chalice/awsclient.py

```diff
@@ -75,10 +78,17 @@ def create_function(self, function_name, role_arn, zip_contents):
                 continue
             return response['FunctionArn']
 
-    def update_function_code(self, function_name, zip_contents):
-        # type: (str, str) -> Dict[str, Any]
-        return self._client('lambda').update_function_code(
+    def update_function(self, function_name, zip_contents,
+                        environment_variables=None):
+        # type: (str, str, Optional[Dict[str, str]]) -> Dict[str, Any]
+        lambda_client = self._client('lambda')
+        return_value = lambda_client.update_function_code(
             FunctionName=function_name, ZipFile=zip_contents)
+        if environment_variables is not None:
```

Comment:
@emellis that makes sense to me as well. The one downside of doing this is that the ``UpdateFunctionConfiguration`` operation will be called every time even when no environment variables were configured. @jamesls is there any other downsides with this approach?